infrasec/tutorials/atlantis.md

```diff
@@ -0,0 +1,418 @@
+
+## Step 7: Troubleshooting 🔧
+
+Due to some bugs in the module and the inherent complexity of integrating/setting up so many resources, some degree of troubleshooting will be necessary. Thus it's included here as a step.
+
+### General IAM Role Assumptions Troubleshooting
+
+We added code to let the Atlantis role control terraform following the  `legendary-waddle` examples for the [s3 backend](https://github.com/trussworks/legendary-waddle/blob/d896c9efb00bf2fb6ca0bf883747852d1851840b/trussworks-misty/atlantis-global/terraform.tf#L10), and the [account provider](https://github.com/trussworks/legendary-waddle/blob/d896c9efb00bf2fb6ca0bf883747852d1851840b/trussworks-misty/admin-global/providers.tf#L3-L5). While we're making changes in terraform for various resources (such as the VPC, ALB, etc.), those resources do not neccesarily also have permissions to control our code. As a result, `terraform init` (as well as any other terraform commands) throw an "access denied" or "unauthorized" error. Temporarily commenting out the assumed role-related code allows us to continue.
+
+### ACM/Certificate Troubleshooting
+
+We'll get a `503` connection refusal error if our certs aren't set up correctly:
+
+<img src="https://github.com/trussworks/Engineering-Playbook/blob/3efe6ea02ed010f3db2c07921c5c8acc60406b84/infrasec/tutorials/images/atlantis_503.png" width="450">
+
+If we get a cert error when we look at our chosen url (in our case `atlantis.exp.move.mil`):
+
+<img src="https://github.com/trussworks/Engineering-Playbook/blob/3efe6ea02ed010f3db2c07921c5c8acc60406b84/infrasec/tutorials/images/atlantis_cert1.png" width="450">
+
+Check the records and corresponding IP addresses in the terminal using `dig move.mil` and `host atlantis.move.mil` to pull up our ACM associated values. Here we also check `orders.exp.move.mil` (corresponding to our misaligned certificate), but it looks similar to this:
+
+![TODO](images/atlantis_cert2.png "TODO")
+
+We can also look at this in Route 53 > Hosted zones check record name --> and look at the `Value/Route traffic to` column. The value should not be going to an IP, it should be going to a CNAME.
+
+We troubleshoot by logging into the console, looking at our ALB, and checking what's associated with it:
+EC2 > load balancer > `atlantis-exp`
+
+<img src="https://github.com/trussworks/Engineering-Playbook/blob/3efe6ea02ed010f3db2c07921c5c8acc60406b84/infrasec/tutorials/images/atlantis_alb2.png" width="450">
+
+We can click on our certificate and manually point to the load balancer.
+
+### VPC Troubleshooting
+
+Anecdotally, the variables for `private_subnets` and `public_subnets` did not succesfully connect to the VPC when plugging in the IP values directly. However, `private_subnet_ids` and `public_subnet_ids` work fine.
+
+There are no open issues in the module for this discrepancy, so this mystery may be confined to the project (or author). This information is included here simply for posterity.
```

Comment:
I don't understand what you're talking about here...

Reply:
It's an anecdotal issue. Was hoping it resonated with someone else.

Reply:
I guess, but this doesn't seem particularly relevant for Atlantis. In a tutorial you really need to keep things focused.

Reply:
This is directly related to a variable in the Atlantis module. It is focused on the Atlantis module.